core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -335,11 +335,13 @@ private void doExecute0(ClientRequestContext ctx, HttpRequestDuplicator rootReqD
                                                  future, derivedCtx, HttpResponse.ofFailure(cause), cause);
                     return null;
                 }
+                derivedCtx.logBuilder().responseContent(null, null);
                 handleResponse(config, ctx, rootReqDuplicator, originalReq, returnedRes, future, derivedCtx,
                                null, aggregated);
                 return null;
             });
         } else {
+            derivedCtx.logBuilder().responseContent(null, null);
```

Comment:
I've set the response content for this issue below.
https://github.com/line/armeria/pull/4827#discussion_r1167421937

According to the `RequestLog`'s javadoc, response content is null for non-RPC. Retry client is for retries of HTTP requests, and so I've put `null` as the response content/preview. @ikhoon nim, may I do like this?
```java
    /**
     * Returns the high-level content object of the {@link Response}, which is specific
     * to the {@link SerializationFormat}.
     *
     * @return {@link RpcResponse} for RPC, or {@code null} for others
     *
     * @throws RequestLogAvailabilityException if the property is not available yet.
     * @see RequestLogProperty#RESPONSE_CONTENT
     */
    @Nullable
    Object responseContent();
```

Reply:
> What do you think of changing the API name to onResponseEnd() and using RequestLogProperty.RESPONSE_END_TIME as the waiting flag?
We need to note that deferred properties may not be available on the callback of rule.onResponseEnd((ctx, log) -> ...)

I got what you meant. Let me update this PR.

Reply:
> We need to note that deferred properties may not be available on the callback of rule.onResponseEnd((ctx, log) -> ...)

How about changing the interface to the following: `rule.onResponseDuration((ctx, nanos) -> {})`? As previously mentioned, we have to make clients aware that some properties may not available. To do that, we'll need to add javadoc or something. Instead, to ensure clarity, we just wait for RequestLogProperty.RESPONSE_END_TIME flag to be set, and pass only the responseDurationNanos rather than the entire requestLog?

Reply:
Only a response duration is exposed by `onResponseDuration((ctx, nanos) -> {})`. I'd like to make it more general to access available information when a response ends.

Anyway, we can simply implement `onResponseDuration(..)` on top of `onResponseEnd()`.
What do you think of adding both versions and getting feedback from other maintainers?

Reply:
I don't think it's a good idea to add `onResponseDuration` because
- we can get the duration easily via the log.
- the duration usually doesn't play a role in Retrying or CircuitBreaking.

I'm also worried about the situation that we end up with a bunch of APIs that are useless.

Reply:
I'm also leaning toward removing the `requestLog` from the `onResponseEnd()` method because
- we can get the `requestLog` easily via `ctx.log()`
- we, anyway, have to check the `RequestLogAvailiability` using `ctx.log().isAvailable(pro)` for the deferred properties.

Reply:
I've changed my mind and I actually like the idea of adding `onResponseDuration((ctx, nanos) -> {...})` instead of `onResponseEnd()`.
We can use the duration when we make a decision for circuit breaking and I think adding `onResponseDuration((ctx, nanos) -> {...})` is much better than adding `onResponseEnd((ctx, log) -> {...})` which uses `RequestLog` for the BiPredicate. `RequestLog` is simple retrievable via `ctx.log()`

Reply:
> What do you think of adding both versions and getting feedback from other maintainers?

@ikhoon nim no problem, but according to @minwoox nim, `onResponseDuration` looks better for him like below. Let me add `onResponseDuration(...)` 🙇 

> I've changed my mind and I actually like the idea of adding onResponseDuration((ctx, nanos) -> {...}) instead of onResponseEnd().
We can use the duration when we make a decision for circuit breaking and I think adding onResponseDuration((ctx, nanos) -> {...}) is much better than adding onResponseEnd((ctx, log) -> {...}) which uses RequestLog for the BiPredicate. RequestLog is simple retrievable via ctx.log()

Thank you for your advice!